website/docs/sql/datatype/array.md

```diff
@@ -0,0 +1,93 @@
+# Array Type
+
+An array is a sequential collection of elements packed into a single SQL value.
+Arrays are useful to model dependent information with an inherent ordering in a single row – for example, time-series data from longitudinal studies, or embedding vectors from machine learning models. 
+
+Generally, array values are denoted in curly braces with elements separated by commas, like so:
+
+```
+{"Lorem", "ipsum", "dolor", "sit", "amed"}
+```
+
+## Element Types and Nullability
+
+Hyper's arrays are _strongly typed_:
+All elements must be of the same type – the array's _element type_.
+The element type is a defining part of the array's overall type, meaning that `array(integer)` is a different type than `array(boolean)` or `array(text)`.
+Arrays can be built from all [atomic types](./index.md) available in Hyper.
+
+Part of an array's element type is the element's nullability. 
+For example, `array(smallint)` is different from `array(smallint not null)`.
+Note that this is independent of the nullability of the array itself.
+The following four options all represent different types in Hyper:
+
+|Type|array nullable?|elements nullable?| possible values|
+|---|---|---|---|
+|`array(integer)`|✅|✅|`{}`,`{1,2,3}`,`{1,2,null}`, `null`|
+|`array(integer not null)`|✅|❌|`{}`,`{1,2,3}`,`null`|
+|`array(integer) not null`|❌|✅|`{}`,`{1,2,3}`,`{1,2,null}`|
+|`array(integer not null) not null`|❌|❌|`{}`,`{1,2,3}`|
+
+The inner nullability of an array type can be changed by casting, using the conventional [cast syntax](../scalar_func/conversion.md):
```

Comment:
Here we should mention which casts are allowed for arrays. I guess it's not only nullability changes, but also casts from and to text.

Reply:
[bda7bea](https://github.com/tableau/hyper-db/pull/110/commits/bda7bea6e20cbb11681f1ca50140f8258c77a809)